On macos 10.15:

```
running: "/nix/store/md0r2kkc0wg2gk18aaim529krci9kv3w-clang-wrapper-11.1.0/bin/c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-arch" "x86_64" "-I" "rocksdb/include/" "-I" "rocksdb/" "-I" "rocksdb/third-party/gtest-1.8.1/fused-src/" "-I" "snappy/" "-I" "." "-Wall" "-Wextra" "-std=c++17" "-Wsign-compare" "-Wsh
  cargo:warning=In file included from rocksdb/cache/fast_lru_cache.cc:10:
  cargo:warning=rocksdb/cache/fast_lru_cache.h:165:3: error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.14 or newer
  cargo:warning=  ~LRUCacheShard() override = default;
  cargo:warning=  ^
  cargo:warning=rocksdb/cache/fast_lru_cache.cc:439:18: note: in defaulted destructor for 'rocksdb::fast_lru_cache::LRUCacheShard' first required here
  cargo:warning=      shards_[i].~LRUCacheShard();
  cargo:warning=                 ^
  cargo:warning=rocksdb/cache/fast_lru_cache.h:165:3: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
  cargo:warning=  ~LRUCacheShard() override = default;
  cargo:warning=  ^
  cargo:warning=1 error generated.
  exit status: 1
```

This will be because we still use the macos 10.12 (?) sdk on x86_64.

--- a/pkgs/servers/search/qdrant/default.nix
+++ b/pkgs/servers/search/qdrant/default.nix
@@ -4,6 +4,7 @@
 , protobuf
 , stdenv
 }:
+
 rustPlatform.buildRustPackage rec {
   pname = "qdrant";
   version = "0.9.1";
@@ -11,7 +12,7 @@ rustPlatform.buildRustPackage rec {
   src = fetchFromGitHub {
     owner = "qdrant";
     repo = "qdrant";
-    rev = "refs/tags/v${version}"; # just v${version} fails to unpack as there is a branch of the same name
+    rev = "refs/tags/v${version}";
     sha256 = "sha256-rOIWiSpAqIUf2V9BMMTZqF/urz754pZV4yHav26dxqY=";
   };
 

--- a/pkgs/servers/search/qdrant/default.nix
+++ b/pkgs/servers/search/qdrant/default.nix
@@ -1,0 +1,34 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, protobuf
+, stdenv
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "qdrant";
+  version = "0.9.1";
+
+  src = fetchFromGitHub {
+    owner = "qdrant";
+    repo = "qdrant";
+    rev = "refs/tags/v${version}"; # just v${version} fails to unpack as there is a branch of the same name
+    sha256 = "sha256-rOIWiSpAqIUf2V9BMMTZqF/urz754pZV4yHav26dxqY=";
+  };
+
+  cargoSha256 = "sha256-ovHxtOYlzVsALw/4bhL9EcFXaKr6Bg/D0w6OPMCLZoQ=";
+
+  nativeBuildInputs = [ protobuf rustPlatform.bindgenHook ];
+
+  NIX_CFLAGS_COMPILE = lib.optional stdenv.isDarwin "-faligned-allocation";
+
+  meta = with lib; {
+    description = "Vector Search Engine for the next generation of AI applications";
+    longDescription = ''
+      Expects a config file at config/config.yaml with content similar to
+      https://github.com/qdrant/qdrant/blob/master/config/config.yaml
+    '';
+    homepage = "https://github.com/qdrant/qdrant";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ dit7ya ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -10185,6 +10185,8 @@ with pkgs;
 
   qarte = libsForQt5.callPackage ../applications/video/qarte { };
 
+  qdrant = callPackage ../servers/search/qdrant { };
+
   qlcplus = libsForQt5.callPackage ../applications/misc/qlcplus { };
 
   qnial = callPackage ../development/interpreters/qnial { };

